### ISSUE TITLE
Re-register does not work

### ISSUE THREAD

**Issue by SBF2978** [andan67/sonytvswitch#1]
Hi andan67,
I had to delete all registered remotes on my Sony KD-AF65 and I wanted to re-register the remote out of the plugin. Until now without any success. The TV is showing a window with the registration number, but there is no window opening up in your plugin.
It's also not possible to delete the remote and add a new one. It finds the ip address automatically, but nothing happens when I click the next button.
There is also no solution after de- and reinstallation of the plugin.
An error message is shown if I try to add a new remote in addition: 
![Screenshot_20200518-222314_TV-Browser vP](https://user-images.githubusercontent.com/61122220/82256451-a3a5a300-9956-11ea-818f-479bf0d60aa9.jpg)

Maybe you can help
Many Thanks in advance

**Comment by andan67** [comment]
Hi,

 

I guess you are using an Android 4.x  device. Correct?

 

The problem is that the latest app version uses a library that is not compatible with Android 4.x, which results in the error message.

Unfortunately, I wasn’t aware of it when I released the app in the store. Otherwise I would have set the minimum Android version accordingly.

I’m currently not sure whether I can fix the issue by using an older version of the external library.

 

For now the only quick solution is to install the previous version from the apk store:

 <https://apkpure.com/de/sony-tv-switch/org.andan.android.tvbrowser.sonycontrolplugin/download/21-APK?from=versions%2Fversion> https://apkpure.com/de/sony-tv-switch/org.andan.android.tvbrowser.sonycontrolplugin/download/21-APK?from=versions%2Fversion

 

This version has one issue that in some circumstances the authentication gets expired (Forbidden error). This can be easily fixed be a re-registration from the Manage control screen.

Please let me know if this works for you.

Please also watch the Google store for further updates.

 

Sorry for the inconvenience and thanks for spotting this bug.

**Comment by SBF2978** [comment]
Hi andan67, no there is android 10 on my S10+ and the TV uses android 8

**Comment by andan67** [comment]
OK. I think I’ve found the bug and will deploy a fixed version shortly.

**Comment by SBF2978** [comment]
I've tried the old version in the meantime and it works. Waiting for your updated  version.
Many thanks for your quick reply.

PS: what I also noticed is that the remote control out of the plugin is not working. The tv is not responding on any input.

**Comment by andan67** [comment]
I’ve triggered the release of a fixed version of the app in the play store which hopefully solves your issue.
If not please let me know.

Re plugin: I’ve noticed on my Samsung device (Android 10) that I needed to kill the TV Browser app to get the plugin work correctly after a new installation or upgrade.

Von: SBF2978
Gesendet: Dienstag, 19. Mai 2020 13:24
An: andan67/sonytvswitch
Cc: andan67; Comment
Betreff: Re: [andan67/sonytvswitch] Re-register does not work (#1)

I've tried the old version in the meantime and it works. Waiting for your updated version.
Many thanks for your quick reply.
PS: what I also noticed is that the remote control out of the plugin is not working. The tv is not responding on any input.

**Comment by SBF2978** [comment]
Will it be version 2.2.1 or how do I know that it is the fixed one?

**Comment by andan67** [comment]
Yes, it will be 2.2.1 with according release notes.


Von: SBF2978
Gesendet: Dienstag, 19. Mai 2020 21:34
An: andan67/sonytvswitch
Cc: andan67; Comment
Betreff: Re: [andan67/sonytvswitch] Re-register does not work (#1)

Will it be version 2.2.1 or how do I know that it is the fixed one?

**Comment by SBF2978** [comment]
Hi andan67,

I updated to the new version and finally it works fine again.
- registration and receipt of the program list works
- then I had to exit the plugin and restart tv browser to match both program lists in your plugin
- after that I wasn't able to switch programs in tv browser, but this was solved after another restart of the app

Many thanks for your quick replies and fast solution. I really appreciate!
